python-connectors/pi-system_event-frames-search/connector.py

```diff
@@ -92,11 +95,16 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                             if not value:
                                 items = batch_row.pop("Items", [])
                                 for item in items:
+                                    batch_row.pop("Links", None)
                                     batch_row.update(item)
+                                    if isinstance(batch_row.get("Value"), dict):
+                                        value = batch_row.pop("Value", {})
+                                        batch_row.update(value)
                                     yield batch_row
                                     if limit.is_reached():
```

Comment:
In batch mode I can't see where limit.add_record() is getting called only - I'm not sure that limit is getting applied to batch mode? Probably I have missed something, I just want to check

Reply:
👍 I simplified it in the latest branch, [f286d91](https://github.com/dataiku/dss-plugin-osisoft/pull/20/commits/f286d916a9d01ea52c07ca2ec7241121379c67af).